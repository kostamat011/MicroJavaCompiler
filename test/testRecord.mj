program testRec

record A {
	int x;
	int y;
	bool b;
}

record B {
	int i;
}

{
	void main() A a; B b; int sum; {
		
		a = new A;
		b = new B;
		
		a.x = 5;
		a.y = a.x * 2;
		a.b = true;
		
		if(a.b) {
			do {
				b.i++;
				a.x = a.x - 1;
			} while(a.x > 0);
		}
		
		print(b.i*2);
	}
}