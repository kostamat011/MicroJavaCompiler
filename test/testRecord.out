   10
Completion took 1 ms